lib/fx/configuration.rb

```diff
@@ -31,8 +32,16 @@ class Configuration
     # @return Fx adapter
     attr_accessor :database
 
+    # Configuration flag that prioritizes the order in the schema.rb of triggers and functions before other statements
+    # in order to make directly schema load work when using functions in statements below.
+    #
+    # Defaults false
+    # @return Bool
+    attr_accessor :define_functions_at_schema_beginning
```

Comment:
This name feels a bit clunky, I don't have a clear suggestion in mind. But here are some wild alternatives:

1) We could expose a `schema_order` argument that would accept `:first` or `:last` that would place the definitions at the top of the `config/schema.rb` file or...at the bottom. I would expect this option to place both functions and triggers at the designated location in the schema.

2) Change this name to be less chunky, and maybe even include triggers? Wild suggestions:
- `define_functions_and_triggers_at_schema_beginning`
- `dump_schema_at_beginning`
- `place_definitions_at_begining_of_schema`
- `place_definitions_at_schema_beginning`

What do you think, do you have any ideas?

Reply:
Yes, it was kind of difficult to find a name for this. I believe that `dump_schema_at_beginning` is the greatest option at the moment, we can still think more about it.

I did not find any problems with triggers at the moment, we required functions defined at the beginning because we are using functions inside indexes and if functions were not defined before indexes, schema loads failed (test suite, setup scripts, etc)

Reply:
I'm thinking something like `dump_definitions_at_beginning_of_schema` is where my mind is at right now - then we include triggers implicitly if we decide later that we want to include them in this feature.